core/project.go

```diff
@@ -0,0 +1,403 @@
+		return "", fmt.Errorf("failed to read schema: %w", err)
+	}
+	return string(outputBytes), nil
+}
+
+func (p *Project) runtime(ctx context.Context, gw bkgw.Client) (*Directory, error) {
+	var runtimeFS *Directory
+	var err error
+	switch p.Config.SDK {
+	case "go":
+		runtimeFS, err = p.goRuntime(ctx, "/", gw, p.Platform)
+	case "python":
+		runtimeFS, err = p.pythonRuntime(ctx, "/", gw, p.Platform)
+	default:
+		return nil, fmt.Errorf("unknown sdk %q", p.Config.SDK)
+	}
+	if err != nil {
+		return nil, err
+	}
+	if _, err := runtimeFS.Stat(ctx, gw, "."); err != nil {
+		return nil, err
+	}
+	return runtimeFS, nil
+}
+
+func (p *Project) getResolvers(ctx context.Context, gw bkgw.Client) (router.Resolvers, error) {
+	resolvers := make(router.Resolvers)
+	if p.Config.SDK == "" {
+		return nil, fmt.Errorf("sdk not set")
+	}
+
+	doc, err := parser.Parse(parser.ParseParams{Source: p.Schema})
+	if err != nil {
+		return nil, err
+	}
+	for _, def := range doc.Definitions {
+		var obj *ast.ObjectDefinition
+
+		if def, ok := def.(*ast.ObjectDefinition); ok {
+			obj = def
+		}
+
+		if def, ok := def.(*ast.TypeExtensionDefinition); ok {
+			obj = def.Definition
+		}
+
+		if obj == nil {
+			continue
+		}
+
+		objResolver := router.ObjectResolver{}
+		resolvers[obj.Name.Value] = objResolver
+		for _, field := range obj.Fields {
+			objResolver[field.Name.Value], err = p.getResolver(ctx, gw)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+	return resolvers, nil
+}
+
+func (p *Project) getResolver(ctx context.Context, gw bkgw.Client) (graphql.FieldResolveFn, error) {
+	runtimeFS, err := p.runtime(ctx, gw)
+	if err != nil {
+		return nil, err
+	}
+
+	return router.ToResolver(func(ctx *router.Context, parent any, args any) (any, error) {
+		pathArray := ctx.ResolveParams.Info.Path.AsArray()
+		name := fmt.Sprintf("%+v", pathArray)
+
+		resolverName := fmt.Sprintf("%s.%s", ctx.ResolveParams.Info.ParentType.Name(), ctx.ResolveParams.Info.FieldName)
+		inputMap := map[string]interface{}{
+			"resolver": resolverName,
+			"args":     args,
+			"parent":   parent,
+		}
+		inputBytes, err := json.Marshal(inputMap)
+		if err != nil {
+			return nil, err
+		}
+		input := llb.Scratch().File(llb.Mkfile(inputFile, 0644, inputBytes))
+
+		fsState, err := runtimeFS.State()
+		if err != nil {
+			return nil, err
+		}
+
+		wdState, err := p.Directory.State()
+		if err != nil {
+			return nil, err
+		}
+
+		st := fsState.Run(
+			llb.Args([]string{entrypointPath}),
+			llb.AddEnv("_DAGGER_ENABLE_NESTING", ""),
+			// make extensions compatible with the shim, in future we can actually enable retrieval of stdout/stderr
+			llb.AddMount("/.dagger_meta_mount", llb.Scratch(), llb.Tmpfs()),
+			llb.AddMount(inputMountPath, input, llb.Readonly),
+			llb.AddMount(tmpMountPath, llb.Scratch(), llb.Tmpfs()),
+		)
+
+		if p.Config.SDK == "go" {
+			st.AddMount("/src", wdState, llb.Readonly) // TODO: not actually needed here, just makes go server code easier at moment
```

Comment:
(Just leaving a bookmark for this TODO in case you forgot, might be intentionally left in)

Reply:
Oh yeah this was a TODO in the original code too. It's gonna be addressed by refactoring the `sdk/go/server.go` code. It currently does all the AST parsing every time it runs, which is incredibly wasteful, but after the refactor I'm planning it will only need to do that once (when called with `-schema`). Then this can be removed.